indexing-service/src/main/java/org/apache/druid/indexing/scheduledbatch/ScheduledBatchTaskManager.java

```diff
@@ -0,0 +1,298 @@
+      final ClientSqlQuery spec
+  )
+  {
+    log.info(
+        "Starting scheduled batch ingestion for supervisorId[%s] with datasource[%s], cronSchedule[%s] and spec[%s].",
+        supervisorId, dataSource, cronSchedulerConfig, spec
+    );
+    final SchedulerManager manager = new SchedulerManager(supervisorId, dataSource, cronSchedulerConfig, spec);
+    manager.startScheduling();
+    supervisorToManager.put(supervisorId, manager);
+  }
+
+  /**
+   * Stops the scheduler for the specified {@code supervisorId}, pausing task submissions
+   * while retaining the ability to track the supervisor's status.
+   */
+  public void stopScheduledIngestion(final String supervisorId)
+  {
+    log.info("Stopping scheduled batch ingestion for supervisorId[%s].", supervisorId);
+    final SchedulerManager manager = supervisorToManager.get(supervisorId);
+    if (manager != null) {
+      // Don't remove the supervisorId from supervisorToManager. We want to be able to track the status
+      // for suspended supervisors to track any in-flight tasks. stop() will clean up all state completely.
+      manager.stopScheduling();
+    }
+  }
+
+  public void start()
+  {
+    log.info("Starting scheduled batch scheduler.");
+    final Optional<TaskRunner> taskRunnerOptional = taskMaster.getTaskRunner();
+    if (taskRunnerOptional.isPresent()) {
+      taskRunnerOptional.get().registerListener(taskRunnerListener, Execs.directExecutor());
+    } else {
+      log.warn("Task runner not registered for scheduled batch scheduler.");
+    }
+  }
+
+  public void stop()
+  {
+    log.info("Stopping scheduled batch scheduler.");
+    final Optional<TaskRunner> taskRunnerOptional = taskMaster.getTaskRunner();
+    if (taskRunnerOptional.isPresent()) {
+      taskRunnerOptional.get().unregisterListener(taskRunnerListener.getListenerId());
+    }
+
+    supervisorToManager.clear();
+  }
+
+  @Nullable
+  public ScheduledBatchSupervisorSnapshot getSchedulerSnapshot(final String supervisorId)
+  {
+    final SchedulerManager manager = supervisorToManager.get(supervisorId);
+    if (manager == null) {
+      return null;
+    }
+
+    final ScheduledBatchStatusTracker.BatchSupervisorTaskStatus tasks = statusTracker.getSupervisorTasks(supervisorId);
+    return new ScheduledBatchSupervisorSnapshot(
+        supervisorId,
+        manager.getSchedulerStatus(),
+        manager.getLastTaskSubmittedTime(),
+        manager.getNextTaskSubmissionTime(),
+        manager.getTimeUntilNextTaskSubmission(),
+        tasks.getSubmittedTasks(),
+        tasks.getCompletedTasks()
+    );
+  }
+
+  private void submitSqlTask(final String supervisorId, final ClientSqlQuery spec)
+      throws ExecutionException, InterruptedException
+  {
+    log.debug("Submitting a new task with spec[%s] for supervisor[%s].", spec, supervisorId);
+    final SqlTaskStatus taskStatus = FutureUtils.get(brokerClient.submitSqlTask(spec), true);
+    statusTracker.onTaskSubmitted(supervisorId, taskStatus);
```

Comment:
Since this is a new feature, we should add an info log with the task ID of the successfully submitted task.